packages/contents/src/drivefs.ts

```diff
@@ -460,10 +486,26 @@ export class ContentsAPI {
   }
 
   private _baseUrl: string;
-  private _driveName: string;
-  private _mountpoint: string;
-  private FS: FS;
-  private ERRNO_CODES: ERRNO_CODES;
+}
+
+/**
+ * An Emscripten-compatible synchronous Contents API using shared array buffers.
+ */
+export class SharedBufferContentsAPI extends ContentsAPI {
+  constructor(driveName: string, mountpoint: string, FS: FS, ERRNO_CODES: ERRNO_CODES) {
+    super(driveName, mountpoint, FS, ERRNO_CODES);
+
+    this.FS = FS;
+    this.ERRNO_CODES = ERRNO_CODES;
+  }
+
+  request(data: IDriveRequest): any {
+    const workerAPI = coincident(self) as typeof globalThis;
+
+    // Assuming the kernel provides a processDriveRequest method
+    // that processes the request with utils.processDriveRequest
+    return (workerAPI as any).processDriveRequest(data);
+  }
```

Comment:
I should probably redesign this to allow the kernel author using whichever tool they want for the blocking communication (coincident, synclink, raw Atomics.wait calls etc...).